execute/plugin.go

```diff
@@ -413,13 +423,13 @@ func (p *Plugin) Outcome(
 			context.Background(),
 			p.lggr,
 			p.msgHasher,
-			p.tokenDataReader,
 			p.reportCodec,
 			p.estimateProvider,
 			observation.Nonces,
```

Comment:
Maybe Nonces should be added to `selectReport` instead of being part of the this constructor.